build/common.nix

```diff
@@ -15,6 +15,14 @@
 
   nixpkgs.config.allowUnfree = true;
 
+  nixpkgs.overlays = [
+    (_prev: final: {
+      # fails to find nix-main against nix 2.28.x
+      # https://github.com/NixOS/infra/pull/620#issuecomment-2784979947
+      nixos-option = final.nixos-option.override { nix = pkgs.nixVersions.nix_2_24; };
+    })
+  ];
+
```

Comment:
N.B. to the future: this can go away in 25.05 when `nixos-option` is just a script that shells out to Nix.